src-java/floodlightrouter-topology/floodlightrouter-storm-topology/src/main/java/org/openkilda/wfm/topology/floodlightrouter/bolts/ControllerToSpeakerProxyBolt.java

```diff
@@ -37,42 +38,60 @@
 import java.util.Set;
 
 @Slf4j
-public class RequestBolt extends AbstractBolt {
-    protected final String outputMessageStream;
-    protected final Set<String> regions;
+public class ControllerToSpeakerProxyBolt extends AbstractBolt {
+    private final String targetTopic;
 
     protected transient SwitchTracker switchTracker;
+    protected final Set<String> allRegions;
 
-    public RequestBolt(String outputMessageStream, Set<String> regions) {
-        this.outputMessageStream = outputMessageStream;
-        this.regions = regions;
+    public ControllerToSpeakerProxyBolt(String targetTopic, Set<String> allRegions) {
+        this.targetTopic = targetTopic;
+        this.allRegions = ImmutableSet.copyOf(allRegions);
     }
 
     @Override
-    public void handleInput(Tuple input) throws Exception {
-        if (Stream.REGION_NOTIFICATION.equals(input.getSourceStreamId())) {
+    protected void dispatch(Tuple input) throws Exception {
+        if (RegionTrackerBolt.STREAM_REGION_UPDATE_ID.equals(input.getSourceStreamId())) {
             handleSwitchMappingUpdate(pullValue(input, AbstractTopology.MESSAGE_FIELD, SwitchMapping.class));
         } else {
-            handleControllerRequest(input);
+            super.dispatch(input);
         }
     }
 
+    @Override
+    public void handleInput(Tuple input) throws Exception {
+        handleControllerRequest(pullControllerPayload(input));
+    }
+
     private void handleSwitchMappingUpdate(SwitchMapping mapping) {
-        updateSwitchMapping(mapping);
+        switchTracker.updateRegion(mapping);
+    }
+
+    private void handleControllerRequest(Object payload) throws PipelineException {
+        if (payload instanceof Message && RouterUtils.isBroadcast((Message) payload)) {
+            handleBroadcastRequest((Message) payload);
+        } else {
+            handleDirectRequest(payload);
```

Comment:
It would be nice to rename `handleDirectRequest` to `handleUnicastRequest` for consistency with BroadcastRequest.

Reply:
done